src/DI/OpenApiPlugin.php

```diff
@@ -43,18 +74,77 @@ public function loadPluginConfiguration()
 		$global = $this->compiler->getExtension()->getConfig();
 		$config = $this->getConfig();
 
+		$infoContact = NULL;
+		if ($config['info']['contact'] !== NULL) {
+			$builder->addDefinition($this->prefix('openapi.info.contact'))
+				->setType(Contact::class)
+				->setFactory(Contact::class, [
+						$config['info']['contact']['name'],
+						$config['info']['contact']['url'],
+						$config['info']['contact']['email'],
+					])
+				->setAutowired(FALSE);
+
+			$infoContact = '@' . $this->prefix('openapi.info.contact');
+		}
+
+		$infoLicense = NULL;
+		if ($config['info']['license'] !== NULL) {
+			$builder->addDefinition($this->prefix('openapi.info.license'))
+				->setType(License::class)
+				->setFactory(License::class, [
+						$config['info']['license'],
+					])
+				->setAutowired(FALSE);
+
+			$infoLicense = '@' . $this->prefix('openapi.info.license');
+		}
+
+		$builder->addDefinition($this->prefix('openapi.info'))
+			->setType(Info::class)
+			->setFactory(Info::class, [
+				$config['info']['title'],
+				$config['info']['description'],
+				$config['info']['termsOfService'],
+				$infoContact,
+				$infoLicense,
+				$config['info']['version'],
+			])
+			->setAutowired(FALSE);
+
+		$schemaTypeFactory = $builder->addDefinition($this->prefix('openapi.schemaType.factory'))
+			->setFactory(SchemaTypeFactory::class);
+
+		foreach ($config['schemaTypes'] as $schemaTypeName => $schemaTypeClass) {
+			$schemaTypeService = $this->prefix('openapi.schemaType.' . $schemaTypeName);
+
+			$builder->addDefinition($schemaTypeService)
+				->setType(ISchemaType::class)
+				->setFactory($schemaTypeClass)
+				->setAutowired(FALSE);
+
+			$schemaTypeFactory->addSetup('$service->addSchemaType(?, ?)', [
```

Comment:
Is it really needed to have service for each schemaType? I vote for 1 service to keep it simple.

Reply:
For me name of `ISchemaType` is matching to `Apitte\Core\Mapping\Parameter\ITypeMapper`. It introduces the possibility to define a Schema based on Openapi (like enums, etc.). So I design it using the same structure.

Reply:
I have thought once again and even I understand your approach, I would prefer to keep it simpler. Who knows maybe in the future, we can make it more complex. For now, it'll be much easier for me to maintain it in 1 class. 

Do you agree?